kube-client/src/client/auth/mod.rs

```diff
@@ -255,6 +273,14 @@ impl TryFrom<&AuthInfo> for Auth {
     fn try_from(auth_info: &AuthInfo) -> Result<Self, Self::Error> {
         if let Some(provider) = &auth_info.auth_provider {
             match token_from_provider(provider)? {
+                #[cfg(feature = "refresh-oidc")]
+                ProviderToken::Oidc(oidc) => {
+                    return Ok(Self::RefreshableToken(RefreshableToken::Oidc(Arc::new(
+                        Mutex::new(oidc),
+                    ))));
+                }
+
+                #[cfg(not(feature = "refresh-oidc"))]
```

Comment:
given that token comes through the auth provider at this point, does it maybe make sense to do this as a runtime check (based on whether there are properties that allow for token refresh) here rather than doing this as a compile time feature check?

Reply:
I think the compile time check is needed here - refreshing flow requires using https. There are two dependencies with https connectors for hyper, but they're both optional

Reply:
that is a good point. it's worth keeping a compile time feature for these dependencies.

my concern here is mainly that users might need to switch configuration of kube to talk to different clusters because one might use oidc with tokens and another might use oidc with refresh. if the `token_from_provider` method identified oidc, it should also figure out "what type of oidc".

There are two error cases currently:
1. the found oidc type requires refresh but the feature is not there
2. the found oidc type does not have refresh, but the feature is enabled
if we had a runtime identification of the oidc type, then we would avoid case 2 being an error. Only case 1 should crash imo.

Reply:
I'm afraid I don't fully understand your concern :v
If the feature is enabled, `token_from_provider` when provided with the minimalistic config (only ID token). The provided ID token is still used until it has expired. After that, if the user tries to use the ID token, we return an error like `your token has expired and you did not configure refreshing properly: ...`

Reply:
Oh, I see. Ok, I didn't realise it would still work in that case. My concern was mostly that I thought the feature was mutually exclusive with the non-refreshable oidc and thus not additive at all. The way we test things is mostly assuming features are additive, so we test with either zero features, default features, or all features.

This is __technically__ two different code-paths for non refreshable oidc tokens (with-or-without the feature), but both are built in CI, so I think this is fine if both paths accomplish the same thing.